apps/server/src/modules/classes/service/class.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
+import { EntityId } from '@shared/domain';
+import { ClassesRepo } from '../repo';
+import { Class } from '../domain';
+
+@Injectable()
+export class ClassService {
+	constructor(private readonly classesRepo: ClassesRepo) {}
+
+	public async deleteUserDataFromClasses(userId: EntityId): Promise<number> {
+		if (!userId) {
+			throw new InternalServerErrorException('User id is missing');
+		}
+
+		const domainObjects = await this.classesRepo.findAllByUserId(userId);
+
+		const updatedClasses: Class[] = domainObjects.map((domainObject) => {
+			if (domainObject.userIds !== undefined) {
+				domainObject.userIds.filter((userId1) => userId1 !== userId);
```

Comment:
since userIds only has a getter, I would not expect this to work. 
It does in this case, but userIds mit as well return a copy to protect itself from changes.

There are a couple of ways to change this. My personal favorite are functions to set data (either updateUsers(), or even removeUsers(), but a setter is possible as well.

Reply:
fixed